src/compression.jl

```diff
@@ -96,12 +86,22 @@ function TranscodingStreams.finalize(codec::ZstdCompressor)
         end
         codec.cstream.ptr = C_NULL
     end
-    reset!(codec.cstream.ibuffer)
-    reset!(codec.cstream.obuffer)
-    return
+    nothing
 end
 
 function TranscodingStreams.startproc(codec::ZstdCompressor, mode::Symbol, error::Error)
+    if codec.cstream.ptr == C_NULL
+        ptr = LibZstd.ZSTD_createCStream()
+        if ptr == C_NULL
+            throw(OutOfMemoryError())
+        end
+        codec.cstream.ptr = ptr
+        i_code = initialize!(codec.cstream, codec.level)
+        if iserror(i_code)
+            error[] = ErrorException("zstd initialization error")
```

Comment:
These errors are unreachable unless there is some allocation error.

Reply:
I'm not sure if even that would reliably trigger an error specifically here, because memory allocations are happening in `ZSTD_createCStream`.